### PR TITLE
Understanding Harvester Resource Metrics Calculation

### PR DESCRIPTION
**Pull request by w13915984028**
https://github.com/harvester/harvester/issues/5012

### CODE REVIEW

File: kb/2024-01-23/reserved_resources_calculation.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
+---
+title: Understanding Harvester Reserved Resources Calculation
+description: Know hwo the resources reseravation is made and calculated
+slug: harvester_reserved_resources_calculation
```
Comment (jillian-maroket) [inline kb/2024-01-23/reserved_resources_calculation.md:4]:
```suggestion
title: Calculation of Reserved Resources in Harvester
description: Understand how reserved resource metrics are calculated.
slug: calculation_of_reserved_resources_in_harvester
```

**Review by jillian-maroket — COMMENTED** [review]
@w13915984028 The draft contained too many headings so I restructured the content and rewrote most of the descriptions to fit the new structure.

File: kb/2024-01-23/reserved_resources_calculation.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+Refer [Longhorn reserved disks](https://longhorn.io/docs/1.5.3/references/settings/#storage-reserved-percentage-for-default-disk) for more details.
+
+Depends on the default disk size, you may adjust this value via [the embedded Longhorn UI](https://docs.harvesterhci.io/v1.2/troubleshooting/harvester/#access-embedded-rancher-and-longhorn-dashboards).
+
+::: note
+
+You are suggested to confirm with Harvester engineering team before change the settings.
+
+:::
```
[the remark this answers is not in the document]
Comment (w13915984028) [reply kb/2024-01-23/reserved_resources_calculation.md]: @jillian-maroket I updated it to `You are suggested to read Longhorn documents carefully before change the settings.`

**Comment by w13915984028** [comment]
@jillian-maroket Thanks all the suggestions, I also amend the document to `harvester resource metrics calculation`, not only the `reserved resources calculation`.  Please take another look, thanks.

File: kb/2024-01-23/harvester_resource_metrics_calculation.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      ephemeral-storage: 153707984Ki
+      hugepages-1Gi: "0"
+      hugepages-2Mi: "0"
+      memory: 20464216Ki
+      pods: "200"
+```
+
+### Resource Usage
+
+CPU and memory usage data is collected by some backend tasks and then stored in `NodeMetrics` object continuously, Harvester reads the data from `usage.cpu` and `usage.memory`.
```
Comment (jillian-maroket) [inline kb/2024-01-23/harvester_resource_metrics_calculation.md:92]:
```suggestion
CPU and memory usage data is continuously collected and stored in the `NodeMetrics` object. Harvester reads the data from `usage.cpu` and `usage.memory`.
```

File: kb/2024-01-23/harvester_resource_metrics_calculation.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+
+### Reserved Storage in Longhorn
+
+Longhorn allows you to specify the percentage of disk space that is not allocated to the default disk on each new Longhorn node. The default value is "30". For more information, see [Storage Reserved Percentage For Default Disk](https://longhorn.io/docs/1.5.3/references/settings/#storage-reserved-percentage-for-default-disk) in the Longhorn documentation.
+
+Depending on the disk size, you can modify the default value using the [embedded Longhorn UI](https://docs.harvesterhci.io/v1.2/troubleshooting/harvester/#access-embedded-rancher-and-longhorn-dashboards).
+
+::: note
+
+You are suggested to read Longhorn documents carefully before change the settings.
```
Comment (jillian-maroket) [inline kb/2024-01-23/harvester_resource_metrics_calculation.md:133]:
```suggestion
Before changing the settings, read the Longhorn documentation carefully.
```

**Review by jillian-maroket — COMMENTED** [review]
@w13915984028 I reviewed the new sentences.

**Review by jillian-maroket — APPROVED** [review]

File: kb/2024-01-23/harvester_resource_metrics_calculation.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    title: Staff Software Engineer
+    url: https://github.com/w13915984028
+    image_url: https://github.com/w13915984028.png
+tags: [harvester, resource metrics, reserved resource, calculation]
+hide_table_of_contents: false
+---
+
+# Calculation of Resource Metrics in Harvester
+
+Harvester calculates the resource using data that is dynamically collected from the system. Host-level resource metrics are calculated and then aggregated to obtain the cluster-level metrics.
```
Comment (markhillgit) [inline kb/2024-01-23/harvester_resource_metrics_calculation.md:16]:
```suggestion
Harvester calculates the resource metrics using data that is dynamically collected from the system. Host-level resource metrics are calculated and then aggregated to obtain the cluster-level metrics.
```

**Review by markhillgit — APPROVED** [review]
I've made one small suggestion at the beginning, but I approve. This looks good and meets the customer requirement, thank you.

File: kb/2024-01-23/harvester_resource_metrics_calculation.md (hunk 5 above)
Comment (w13915984028) [reply kb/2024-01-23/harvester_resource_metrics_calculation.md]: Updated.
Thanks @markhillgit

**Review by bk201 — APPROVED** [review]

**Pull request merged**